exercises/exercise4/checkSpecialCases.xquery

```diff
@@ -0,0 +1,149 @@
+xquery version "1.0";
+
+declare namespace local = "local";
+declare variable $gameInstance := fn:doc("GameInstance.xml")/mancalaGame;
+declare variable $currentPlayer := local:getCurrentPlayer();
+declare variable $oppositePit := local:getOppositePit($lastPit);
+
+(: This is the main function which triggers all other helper functions :)
+declare function local:checkSpecialCases($lastPit as xs:string) {
+    (: Check if last pit is from current player AND last pit was empty before AND opposite pit has any seeds? :)
+    if (local:getIsPitOfCurrentPlayer($lastPit, $currentPlayer) and local:getWasLastPitEmpty($lastPit) and local:getOppositePitHasSeeds($oppositePit)) then (
+        (: All checks have passed, move last pit and opposite pit seeds to current players store :)
+        local:fillPlayersStore(local:getCurrentPlayersStore($currentPlayer)) and local:setPitsToZero($lastPit)
+    )
+    else (
```

Comment:
Special case 2 still has to be checked. Namely if the last lands into the players own store.